Cargo.toml

```diff
@@ -48,6 +48,10 @@ libtock_temperature = { path = "apis/temperature" }
 
 embedded-hal = { version = "1.0", optional = true }
 
+mipidsi = "0.8.0"
```

Comment:
I agree. @jrvanwhy and I talked about this in the OT call a little bit today.

Obviously this doesn't work, we need to figure out how to handle the features and dependencies. Either through some Cargo magic or a separate sub crate (or any other ideas)

Reply:
Here are a few ideas on how we can handle this.

# Idea A: Make this example its own crate.

What Amit said. For comparison with the others, here's a more thorough description:

Move this one out of the examples/ directory into a new cargo package. Say `complex_examples/` (I'm not very happy with the name, feel free to change). Make it the one example that's different from the rest. Tweak the makefile so that e.g. `make examples` builds the new example crate in addition to the examples that Cargo autodetects.

# Idea B: `[dev-dependencies]` and `[[example]].required-features`

This is the easiest option to implement (i.e. the smallest delta). It has two parts. First, to avoid adding the new dependencies to all of `libtock`'s users, move the new dependencies into `[dev-dependencies]`:

```toml
[dev-dependencies]
display-interface-api = "0.5"
embedded-graphics = "0.8"
mipidsi = "0.8.0"
```

Then, to avoid the compilation error, tell Cargo to only build the `st7789` example if the `rust_embedded` feature is enabled:

```toml
[[example]]
name = "st7789"
required-features = ["rust_embedded"]
```

Note that this does not automatically enable the rust_embedded feature, so running `make examples` (and similar commands) will skip building the st7789 example. We could leave it as-is or tweak `make examples` to enable additional features (maybe just have it turn them all on).

# Idea C: make `examples/` its own crate.

Using `autoexamples = false`, I think we can make `examples/` a normal crate with a number of `bin` targets inside. It would depend on `libtock` with `features = ["rust_embedded"]`, so building it would automatically turn on all the features.

This would probably require some refactoring of the makefile, and `cargo` would no longer understand that these binaries are all examples.

# Idea D: make `examples/` a collection of crates.

This is basically the same as idea C, except instead of having one example crate, we make every example its own crate. Then only the st7789 crate would depend on the rust_embedded feature and the additional dependencies.

Reply:
I have implemented B in this PR.

I'm working on A as well, but it doesn't seem to link correctly. The WIP commit is here: https://github.com/tock/libtock-rs/commit/3e57563651f05eee929b161b8759595a6ad2fa35